Model/Transaction.php

```diff
@@ -398,4 +405,41 @@ public function getMessages()
     {
         return new ArrayCollection($this->messages);
     }
+
+    /**
+     * ResponseTime setter
+     *
+     * @param int $milliseconds
+     *
+     * @return Transaction
+     */
+    public function setResponseTime($milliseconds)
+    {
+        $this->responseTime = $milliseconds;
+        return $this;
+    }
+
+    /**
+     * ResponseTime getter
+     *
+     * @return int
+     */
+    public function getResponseTime()
+    {
+        return $this->responseTime;
+    }
+
+    /**
+     * Calculate response time in milliseconds based on current and start of request timestamps
+     *
+     * @param float $currentTimestamp
+     *
+     * @return int
+     */
+    public function calculateResponseTime($currentTimestamp)
+    {
+        $responseTime = round(($currentTimestamp - $_SERVER['REQUEST_TIME_FLOAT']) * 1000);
```

Comment:
You should be getting REQUEST_TIME from the Symfony native component. See https://symfony.com/doc/current/components/http_foundation.html

Also, this calculation should be happening in a service and not in the model. A transaction model should not know anything about server values, it should only be concerned about the current transaction.